lab4/src/map.py

```diff
@@ -0,0 +1,73 @@
+from typing import List
+from .country import Country
+from .city import City
+
+
+class Map:
+    def __init__(self, countries_data):
+        self.countries = []
+        self.grid: List[List[City]] = [[None] * (10 + 2) for _ in range((10 + 2))]
+        self.__initialize_grid(countries_data)
+        self.__validate_foreign_neighbours()
+
+    def simulate_euro_diffusion(self) -> None:
+        if len(self.countries) == 1:
+            self.countries[0].day_of_full = 0
+            return
+
+        full = False
+        day = 1
+        while not full:
+            for x in range(10 + 1):
+                for y in range(10 + 1):
```

Comment:
Remove "magic" numbers, move them to separate variables.

Reply:
Move to private variable _grid_offset